File: src/my_class_pkg/scripts/ros_client.py
#!/usr/bin/env python3
import rospy
from my_class_pkg.srv import MyServiceMsg

if __name__ == "__main__":
    rospy.init_node("my_service_client")
    rospy.wait_for_service("my_service")
    client = rospy.ServiceProxy("my_service", MyServiceMsg)
    try:
        resp = client.call(2)
        rospy.loginfo("Service response: %d", resp.output)
    except rospy.ServiceException as e:
        rospy.logerr("Service call failed: %s", e)

File: src/my_class_pkg/src/ros_client.cpp
#include "ros/ros.h"
#include "my_class_pkg/MyServiceMsg.h"

int main(int argc, char **argv)
{
    ros::init(argc, argv, "my_service_client");
    ros::NodeHandle nh;
    ros::ServiceClient client = nh.serviceClient<my_class_pkg::MyServiceMsg>("my_service");
    my_class_pkg::MyServiceMsg srv;
    srv.request.input = 42;
    if (client.call(srv))
    {
        ROS_INFO("Service response: %ld", (long)srv.response.output);
    }
    else
    {
        ROS_ERROR("Failed to call service my_service");
        return 1;
    }
    return 0;
}
